src/System.ComponentModel.TypeConverter/src/System/ComponentModel/VersionConverter.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    ///    <para>Provides a type converter to convert Version objects to and
+    ///       from various other representations.</para>
+    /// </summary>
+    public class VersionConverter : TypeConverter
+    {
+        /// <summary>
+        ///    <para>Gets a value indicating whether this converter can convert an object in the
+        ///       given source type to a Version.</para>
+        /// </summary>
+        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
+        {
+            if (sourceType == null)
+                throw new ArgumentNullException(nameof(sourceType));
+
+            return sourceType == typeof(string) || sourceType == typeof(Version);
+        }
+
+        /// <summary>
+        ///    <para>Gets a value indicating whether this converter can
+        ///       convert an object to the given destination type using the context.</para>
+        /// </summary>
+        public override bool CanConvertTo(ITypeDescriptorContext context, Type destinationType)
+        {
+            return destinationType == typeof(string) || destinationType == typeof(Version);
+        }
+
+        /// <summary>
+        ///    <para>Converts the given object to a Version.</para>
+        /// </summary>
+        public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, object value)
+        {
+            if (value is string versionString)
+            {
+                // Let the Version constructor throw any informative exceptions
+                return new Version(versionString);
+            }
+
+            if (value is Version version)
+            {
+                return new Version(version.Major, version.Minor, version.Build, version.Revision); // return new instance
+            }
+
+            throw GetConvertFromException(value);
```

Comment:
Other TypeConverters appear to delegate to the base implementation (which in turn calls this).  We should follow the same pattern unless there's a really good reason not to.

Reply:
You're right, there is no good reason, I addressed this in a5ad73cc1ebfbec15ca1d5cdc007cf72654344fe.

The origin of throwing directly is that I used `UriTypeConverter` as a model to implement `VersionConverter` and `UriTypeConverter` throws directly without calling the base class implementation.